### PR TITLE
bpf: ct: clean up redundant 0-initializiations for CT entry creation

### PR DESCRIPTION
**Pull request by julianwiedmann**
Follow-up for https://github.com/cilium/cilium/pull/30344/commits/7abe5e7c3cdccafa97223fc0c10d70e1ef95ba55:

>Note that this introduces a few additional explicit initializations of the
fields in ct_state, and we can potentially avoid some of them. But this
in-detail investigation is best done in a later patch.

Let's do this now :).

### CODE REVIEW

**Comment by julianwiedmann** [comment]
(CI is still recovering from the GH outage)

File: bpf/lib/host_firewall.h [hunk 1]
```diff
@@ -125,9 +124,10 @@ __ipv6_host_policy_egress(struct __ctx_buff *ctx, bool is_host_id __maybe_unused
 
 	/* Only create CT entry for accepted connections */
 	if (ret == CT_NEW && verdict == CTX_ACT_OK) {
+		struct ct_state ct_state_new = {};
```
Comment (aspsk) [inline bpf/lib/host_firewall.h:127]: or
```
struct ct_state ct_state_new = {
        .src_sec_id = HOST_ID,
        .proxy_redirect = proxy_port > 0,
};
```

**Review by aspsk — APPROVED** [review]
lgtm, one cosmetic comment

File: bpf/lib/host_firewall.h (hunk 1 above)
Comment (julianwiedmann) [reply bpf/lib/host_firewall.h:127]: :nod: but it's not a pattern we typically follow I think. Luckily makes no difference to the compiler.

**Pull request merged**